inlong-manager/manager-client/src/main/java/org/apache/inlong/manager/client/api/util/InlongStreamSinkTransfer.java

```diff
@@ -244,21 +230,32 @@ private static HiveSinkRequest createHiveRequest(StreamSink streamSink, InlongSt
         hiveSinkRequest.setPrimaryPartition(hiveSink.getPrimaryPartition());
         hiveSinkRequest.setSecondaryPartition(hiveSink.getSecondaryPartition());
         if (CollectionUtils.isNotEmpty(hiveSink.getStreamFields())) {
-            List<SinkFieldRequest> fieldRequests = hiveSink.getStreamFields()
-                    .stream()
-                    .map(streamField -> {
-                        SinkFieldRequest storageFieldRequest = new SinkFieldRequest();
-                        storageFieldRequest.setFieldName(streamField.getFieldName());
-                        storageFieldRequest.setFieldType(streamField.getFieldType().toString());
-                        storageFieldRequest.setFieldComment(streamField.getFieldComment());
-                        return storageFieldRequest;
-                    })
-                    .collect(Collectors.toList());
+            List<SinkFieldRequest> fieldRequests = createSinkFields(streamSink.getStreamFields(),
+                    streamInfo.getFieldList());
             hiveSinkRequest.setFieldList(fieldRequests);
         }
         return hiveSinkRequest;
     }
 
+    private static List<SinkFieldRequest> createSinkFields(List<StreamField> sinkFields,
+            List<InlongStreamFieldInfo> sourceFields) {
+        List<SinkFieldRequest> sinkFieldRequests = Lists.newArrayList();
+        for (int i = 0; i < sinkFields.size(); i++) {
+            SinkFieldRequest sinkFieldRequest = new SinkFieldRequest();
+            StreamField sinkField = sinkFields.get(i);
+            sinkFieldRequest.setFieldName(sinkField.getFieldName());
+            sinkFieldRequest.setFieldType(sinkField.getFieldType().toString());
+            sinkFieldRequest.setFieldComment(sinkField.getFieldComment());
+            if (sourceFields.size() > i && sourceFields.get(i) != null) {
+                InlongStreamFieldInfo sourceField = sourceFields.get(i);
+                sinkFieldRequest.setSourceFieldName(sourceField.getFieldName());
+                sinkFieldRequest.setSourceFieldType(sourceField.getFieldType());
+            }
+            sinkFieldRequests.add(sinkFieldRequest);
+        }
+        return sinkFieldRequests;
+    }
+
```

Comment:
Is field mapping created by order of sink field list and stream field list ?

Reply:
It is now, and yunqingmo will add Mappings after this pr is merged